cpp/include/raft/comms/detail/std_comms.hpp

```diff
@@ -103,12 +103,14 @@ class std_comms : public comms_iface {
     initialize();
   };
 
-  void initialize()
+  ~std_comms()
   {
-    sendbuff_ = status_.data();
-    recvbuff_ = status_.data() + 1;
+    requests_in_flight_.clear();
+    free_requests_.clear();
   }
 
+  void initialize() { sendbuff_ = status_.data(); }
```

Comment:
May better set status_ to 0 to avoid status_ holding garbage. Unlikely to affect any program behavior, but I guess this is a safer practice (especially considering that performance overhead is negligible).

And if we use rmm::device_scalar instead of rmm::device_uvector, we can simply call set_value_to_zero_async (https://github.com/rapidsai/rmm/blob/branch-22.04/include/rmm/device_scalar.hpp#L215).